include/EASTL/tuple.h

```diff
@@ -145,6 +145,41 @@ class tuple_element<I, const volatile Internal::TupleImpl<Indices, Ts...>> : pub
 {
 };
 
+// attempt to isolate index given a type
+template <typename T, typename Tuple>
+struct tuple_index
+{
+};
+
+template <typename T>
+struct tuple_index<T, TupleTypes<>>
+{
+	typedef void DuplicateTypeCheck;
+	tuple_index() = delete; // tuple_index should only be used for compile-time assistance, and never be instantiated
+	static const size_t index = 0;
+};
+
+template <typename T, typename... Ts>
+struct tuple_index<T, TupleTypes<T, Ts...>>
+{
+	typedef int DuplicateTypeCheck;
+	static_assert(is_void<typename tuple_index<T, TupleTypes<Ts...>>::DuplicateTypeCheck>::value, "duplicate type T in tuple_vector::get<T>(); unique types must be provided in declaration, or only use get<size_t>()");
+
+	static const size_t index = 0;
+};
+
+template <typename T, typename Ts, typename... TsRest>
```

Comment:
I'd rename Ts to TsHead to clearly show that you are taking the head of the type list and then recursing further down the type list.  This will help to differentiate it from T which is the requested type.